mocha-multi.js

```diff
@@ -35,8 +35,18 @@ process.exit = function(code) {
   process.on('exit', quit);
 }
 
-function MochaMulti(runner) {
-  var streams = initReportersAndStreams(runner, parseSetup());
+function MochaMulti(runner, options) {
+  var setup=[];
+  if( options && options.reporterOptions ) {
+    debug("options %j", options.reporterOptions);
+    var reporters=Object.keys(options.reporterOptions);
+    reporters.forEach(function(reporter) {
+      debug("%j %j", options.reporterOptions, reporter);
+      setup.push([ reporter, options.reporterOptions[reporter].stdout, options.reporterOptions[reporter].options ]);
+    });
+  }
+  debug("setup %j", setup);
+  var streams = initReportersAndStreams(runner, setup || parseSetup());
```

Comment:
You're right, thanks. I made a fix and the unit tests pass now. I've pushed
my changes to my fork, would you like a new PR?

On Thu, Apr 16, 2015 at 3:23 AM, Glen Mailer <notifications@github.com>
wrote:

> In mocha-multi.js
> <https://github.com/glenjamin/mocha-multi/pull/9#discussion_r28498531>:
>
> > @@ -35,8 +35,18 @@ process.exit = function(code) {
> >    process.on('exit', quit);
> >  }
> >
> > -function MochaMulti(runner) {
> > -  var streams = initReportersAndStreams(runner, parseSetup());
> > +function MochaMulti(runner, options) {
> > +  var setup=[];
> > +  if( options && options.reporterOptions ) {
> > +    debug("options %j", options.reporterOptions);
> > +    var reporters=Object.keys(options.reporterOptions);
> > +    reporters.forEach(function(reporter) {
> > +      debug("%j %j", options.reporterOptions, reporter);
> > +      setup.push([ reporter, options.reporterOptions[reporter].stdout, options.reporterOptions[reporter].options ]);
> > +    });
> > +  }
> > +  debug("setup %j", setup);
> > +  var streams = initReportersAndStreams(runner, setup || parseSetup());
>
> Ah, the problem is here - setup is never falsy, so it will never use the
> environment variable.
>
> Perhaps do something like:
>
> var setup;if (options && options.reporterOptions) {
>   setup = // ...
> } else {
>   setup = parseSetup()
> }
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/glenjamin/mocha-multi/pull/9/files#r28498531>.
>



-- 
We should look for someone to eat and drink with before looking for
something to eat and drink... ~Epicurus